catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/ShareableMetacardPolicyExtension.java

```diff
@@ -106,17 +115,17 @@ private KeyValueCollectionPermission isPermitted(
       return match; // ignore all but shareable permissions
     }
 
-    Predicate<CollectionPermission> isSystem = system();
-    Predicate<CollectionPermission> isOwner = owner(grouped);
-    Predicate<CollectionPermission> hasAccessIndividuals = individuals(grouped);
-    Predicate<CollectionPermission> hasAccessGroups = groups(grouped);
+    boolean isSystem = system(subject);
+    boolean isOwner = owner(subject, grouped);
+    boolean hasAccessIndividuals = individuals(subject, grouped);
+    boolean hasAccessGroups = groups(subject, grouped);
 
     // get all permissions implied by the subject
     Supplier<Set<String>> impliedPermissions =
```

Comment:
While you are in the code, you can also remove this supplier.

Reply:
Yeah if we don't want to wait for the subsequent PR I can do additionally rework now if desired.